### PR TITLE
Index sources

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: search/engine.go [hunk 1]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+		sources_indices[i] = es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i])
+		log.Infof("adding index to sources_indices: %s", es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i]))
+	}
+	fetchSourceContext := elastic.NewFetchSourceContext(true).
+		Include("mdb_uid")
+	for _, index := range sources_indices {
+		searchSource := elastic.NewSearchSource().
+			Query(createSourcesQuery(query)).
+			Highlight(elastic.NewHighlight().Fields(
+				elastic.NewHighlighterField("name"),
```
Comment (bbfsdev) [inline search/engine.go]: `elastic.NewHighlighterField("name").NumOfFragments(0),`
This say that when you highlight you show full name.

File: search/engine.go [hunk 2]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+				elastic.NewHighlighterField("description"),
+				elastic.NewHighlighterField("authors"),
+				elastic.NewHighlighterField("content"),
+				elastic.NewHighlighterField("name.analyzed"),
```
Comment (bbfsdev) [inline search/engine.go]: NumOfFragments(0)

File: search/engine.go [hunk 3]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+				elastic.NewHighlighterField("description.analyzed"),
+				elastic.NewHighlighterField("authors.analyzed"),
```
Comment (bbfsdev) [inline search/engine.go]: NumOfFragments(0)

File: search/engine.go [hunk 4]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+	}
+	fetchSourceContext := elastic.NewFetchSourceContext(true).
+		Include("mdb_uid")
+	for _, index := range sources_indices {
+		searchSource := elastic.NewSearchSource().
+			Query(createSourcesQuery(query)).
+			Highlight(elastic.NewHighlight().Fields(
+				elastic.NewHighlighterField("name"),
+				elastic.NewHighlighterField("description"),
+				elastic.NewHighlighterField("authors"),
```
Comment (bbfsdev) [inline search/engine.go]: NumOfFragments(0)

File: search/engine.go [hunk 5]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+	for i := range query.LanguageOrder {
+		sources_indices[i] = es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i])
+		log.Infof("adding index to sources_indices: %s", es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i]))
+	}
+	fetchSourceContext := elastic.NewFetchSourceContext(true).
+		Include("mdb_uid")
+	for _, index := range sources_indices {
+		searchSource := elastic.NewSearchSource().
+			Query(createSourcesQuery(query)).
+			Highlight(elastic.NewHighlight().Fields(
```
Comment (bbfsdev) [inline search/engine.go]: Should be `Highlight(elastic.NewHighlight().HighlighterType("unified").Fields(`

File: search/engine.go [hunk 6]
```diff
@@ -360,10 +437,10 @@ func compareHits(h1 *elastic.SearchHit, h2 *elastic.SearchHit, sortBy string) (b
 
 func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy string, from int, size int) (*elastic.SearchResult, error) {
 	log.Infof("%+v %+v", r1, r2)
-	if r1.Hits.TotalHits == 0 {
+	if r1 == nil || r1.Hits == nil || r1.Hits.TotalHits == 0 {
```
Comment (bbfsdev) [inline search/engine.go]: Create function similar to haveHits that is called noHits and use it here.

File: search/engine.go [hunk 7]
```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
...
+
+		if indexesCount == 1 || indexesCount == 3 {
+			sR = mr.Responses[i+(len(query.LanguageOrder)*(indexesCount-1))]
+			if sR.Error != nil {
+				log.Warnf("%+v", sR.Error)
+				return nil, errors.New("Failed multi get.")
+			}
 		}
+
 		if haveHits(cuR) || haveHits(cR) {
```
Comment (bbfsdev) [inline search/engine.go]: You don't check here `haveHits(sR)`

File: search/engine.go [hunk 8]
```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
...
-			log.Debugf("Joining:\n%+v\n\n%+v\n\n\n", cuR.Hits, cR.Hits)
-			ret, err := joinResponses(cuR, cR, sortBy, from, size)
-			log.Debugf("Res: %+v", ret.Hits)
-			return ret, err
+			joinedR, err = joinResponses(cuR, cR, sortBy, from, size)
+			if err != nil {
+				return joinedR, err
+			}
 		}
+		joinedR, err = joinResponses(joinedR, sR, sortBy, from, size)
```
Comment (bbfsdev) [inline search/engine.go]: Rename `joinReponses` to `joinTwoResponses`
Implement new joinResponses function as variadic:
```
func joinResponses(sortBy string, from int, size int, results... *elastic.SearchResult)) {
 ...
   use somewhere here joinTwoResponses and comment to refactor it later.
 ...
}
Within `joinReponses` `results` will be []*elastic.SearchResult{cR, cuR, sR}

joinedR, err = joinResponses(sortBy, from, size, cuR, cR, sR)
```

File: search/engine.go [hunk 9]
```diff
@@ -406,10 +483,30 @@ func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy st
...
 	log.Infof("Query: %+v sort by: %s, from: %d, size: %d", query, sortBy, from, size)
 	multiSearchService := e.esc.MultiSearch()
-	// Content Units
-	AddContentUnitsSearchRequests(multiSearchService, query, sortBy, 0, from+size, preference)
-	// Collections
-	AddCollectionsSearchRequests(multiSearchService, query, sortBy, 0, from+size, preference)
+
+	var indexesCount int
+	if len(query.Filters) == 0 {
+		log.Info("No section selected. Searching in all indexes.")
```
Comment (bbfsdev) [inline search/engine.go]: No need to log this.

File: search/engine.go [hunk 10]
```diff
@@ -406,10 +483,30 @@ func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy st
...
+		indexesCount = 3
+	} else if _, ok := query.Filters[consts.FILTERS[consts.FILTER_SECTION_SOURCES]]; ok {
+		log.Info("Sources section selected. Searching only in sources.")
```
Comment (bbfsdev) [inline search/engine.go]: No need to log this.

File: search/engine.go [hunk 11]
```diff
@@ -406,10 +483,30 @@ func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy st
...
+		indexesCount = 1
+	} else {
+		log.Info("Non sources section selected. Searching only in content units and collections.")
```
Comment (bbfsdev) [inline search/engine.go]: No need to log this.

File: search/engine.go [hunk 12]
```diff
@@ -406,10 +483,30 @@ func joinResponses(r1 *elastic.SearchResult, r2 *elastic.SearchResult, sortBy st
...
+		indexesCount = 2
+	}
+
+	if indexesCount >= 2 {
```
Comment (bbfsdev) [inline search/engine.go]: This is not the right way to handle this we have 3 states:

NO_FILTER
ONLY_SOURCES
WITHOUT_SOURCES

You need an enum with those values. Also I have noticed that in the client we should consider disabling non-valid filtering...

Comment (bbfsdev) [reply search/engine.go]:
```
requests = make([]...)
if NO_FILTER || WITHOUT_SOURCES
  requests = append(requests,
      content units,
      collections)  // I think append is variadic

if NO_FILTER || ONLY_SOURCES
  requests = append(requests, sources)
```

File: search/engine.go [hunk 13]
```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		return nil, errors.Wrap(err, "ES error.")
 	}
 
-	if len(mr.Responses) != 2*len(query.LanguageOrder) {
+	if len(mr.Responses) != indexesCount*len(query.LanguageOrder) {
```
Comment (bbfsdev) [inline search/engine.go]: Here you should use slice of added search requests and use it's length.

File: search/engine.go [hunk 14]
```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
...
-		if cuR.Error != nil {
-			log.Warnf("%+v", cuR.Error)
-			return nil, errors.New("Failed multi get.")
+
+		var cuR *elastic.SearchResult
+		var cR *elastic.SearchResult
+		var sR *elastic.SearchResult
+		var joinedR *elastic.SearchResult
+
+		if indexesCount >= 2 {
```
Comment (bbfsdev) [inline search/engine.go]: Use only the len(requests) here and make this code generic.
Don't split to:
```
var cuR *elastic.SearchResult
var cR *elastic.SearchResult
var sR *elastic.SearchResult
```

instead use
`var results []*elastic.SearchResult`

File: search/engine.go (hunk 7 above)
Comment (bbfsdev) [reply search/engine.go]: Create haveAnyHits variadic function

File: es/common.go [hunk 15]
```diff
@@ -17,6 +18,7 @@ import (
 )
 
 var (
+	SourcesFolder   string
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/common.go]: fixed

File: search/engine.go (hunk 6 above)
Comment (YuriHechter) [reply search/engine.go]: using !haveHits

File: es/common.go (hunk 15 above)
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/common.go]: used by indexer_tests

File: es/content_units_index.go [hunk 16]
```diff
@@ -385,7 +385,9 @@ func (index *ContentUnitsIndex) indexUnit(cu *mdbmodels.ContentUnit, indexData *
 							log.Errorf("Error downloading or converting doc: %s", val[0])
 							log.Errorf("Error %+v", err)
 						} else {
-							unit.Transcript, err = index.parseDocx(val[0])
+							docxFilename := fmt.Sprintf("%s.docx", val[0])
+							docxPath := path.Join(docFolder, docxFilename)
+							unit.Transcript, err = index.ParseDocx(docxPath)
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/content_units_index.go]: deleted

File: es/indexer_test.go [hunk 17]
```diff
@@ -1134,3 +1405,54 @@ func (suite *IndexerSuite) TestCollectionsScopeByFile() {
...
+
+	fmt.Println("Validate source files.")
+	suite.validateSourceFile(indexNameEn, indexer, "TEST CONTENT")
+	suite.validateSourceFile(indexNameHe, indexer, "TEST CONTENT")
+
+	fmt.Println("Validate adding source incrementally.")
+	source2UID := suite.us(es.Source{Name: "test-name-2", Description: "test-description-2"}, consts.LANG_ENGLISH)
+	suite.usfc(source2UID, consts.LANG_ENGLISH)
+	r.Nil(indexer.SourceAdd(source2UID))
+	suite.validateSourceNames(indexNameEn, indexer, []string{"test-name-1", "test-name-2"})
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: validation added

File: es/indexer_test.go [hunk 18]
```diff
@@ -1134,3 +1405,54 @@ func (suite *IndexerSuite) TestCollectionsScopeByFile() {
...
+
+	fmt.Println("Add 2 authors and validate.")
+	suite.asa(es.Source{MDB_UID: source1UID}, consts.LANG_ENGLISH, mdbmodels.Author{Name: "Test Name", ID: 3, Code: "t1", FullName: null.String{String: "Test Full Name", Valid: true}}, true, true)
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: @edoshor

File: es/sources_index.go [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+type SourcesIndex struct {
+	BaseIndex
+}
+
+func (index *SourcesIndex) ReindexAll() error {
+	log.Infof("Sources Index - Reindex all.")
+	if _, err := index.removeFromIndexQuery(elastic.NewMatchAllQuery()); err != nil {
+		return err
+	}
+	return index.addToIndexSql("1=1")
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: done

File: es/sources_index.go [hunk 20]
```diff
@@ -0,0 +1,273 @@
...
+			qm.Limit(limit)).Bind(&sources)
+		if err != nil {
+			return errors.Wrap(err, "Fetch sources from mdb")
+		}
+		log.Infof("Adding %d sources (offset: %d).", len(sources), offset)
+
+		for _, source := range sources {
+			if err := index.indexSource(source); err != nil {
+				log.Warnf("*** Unable to index source '%s' (uid: %s). Error is: %v. ***", source.Name, source.UID, err)
+				//return err
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: indexSource() has been refactored. Now returning error.

File: es/sources_index.go [hunk 21]
```diff
@@ -0,0 +1,273 @@
...
+			qm.Offset(offset),
+			qm.Limit(limit)).Bind(&sources)
+		if err != nil {
+			return errors.Wrap(err, "Fetch sources from mdb")
+		}
+		log.Infof("Adding %d sources (offset: %d).", len(sources), offset)
+
+		for _, source := range sources {
+			if err := index.indexSource(source); err != nil {
+				log.Warnf("*** Unable to index source '%s' (uid: %s). Error is: %v. ***", source.Name, source.UID, err)
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: removed ***

File: es/sources_index.go [hunk 22]
```diff
@@ -0,0 +1,273 @@
...
+			}
+
+			if i18n.Description.Valid && i18n.Description.String != "" {
+				source.Description = i18n.Description.String
+			}
+
+			fPath, err := index.getDocxPath(mdbSource.UID, i18n.Language)
+			if err != nil {
+				log.Warnf("Unable to retrieving docx path for source %s with language %s", mdbSource.UID, i18n.Language)
+				continue
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: log updated: "skipping indexing"

File: es/sources_index.go [hunk 23]
```diff
@@ -0,0 +1,273 @@
...
+		return []string{}, nil
+	}
+	keys := make([]string, 0)
+	for k := range removed {
+		keys = append(keys, k)
+	}
+	return keys, nil
+}
+
+func (index *SourcesIndex) getDocxPath(uid string, lang string) (string, error) {
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: it is not the same

File: es/sources_index.go [hunk 24]
```diff
@@ -0,0 +1,273 @@
...
+				if ai18n.FullName.Valid && ai18n.FullName.String != "" {
+					source.Authors = append(source.Authors, ai18n.FullName.String)
+				}
+			}
+
+			i18nMap[i18n.Language] = source
+		}
+	}
+
+	if !hasDocxForSomeLanguage {
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: this part has been refactored

File: search/engine.go [hunk 25]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+			elastic.NewConstantScoreQuery(
+				elastic.NewBoolQuery().Should(
+					elastic.NewMatchQuery("name.analyzed", q.Term),
+					elastic.NewMatchQuery("description.analyzed", q.Term),
+					elastic.NewMatchQuery("content.analyzed", q.Term),
+					elastic.NewMatchQuery("authors.analyzed", q.Term),
+				).MinimumNumberShouldMatch(1)).Boost(1),
+		).Should(
+			elastic.NewBoolQuery().Should(
+				elastic.NewMatchPhraseQuery("name.analyzed", q.Term).Slop(100).Boost(1.5),
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go [hunk 26]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+				elastic.NewMatchPhraseQuery("description.analyzed", q.Term).Slop(100).Boost(1.2),
+				elastic.NewMatchPhraseQuery("content.analyzed", q.Term).Slop(100),
+				elastic.NewMatchPhraseQuery("authors.analyzed", q.Term).Slop(100),
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply search/engine.go]: 1.2 changed to 1.5

File: search/engine.go [hunk 27]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+			elastic.NewBoolQuery().Should(
+				elastic.NewMatchPhraseQuery("name", exactTerm).Slop(100).Boost(1.5),
+				elastic.NewMatchPhraseQuery("description", exactTerm).Slop(100).Boost(1.2),
+				elastic.NewMatchPhraseQuery("content.analyzed", exactTerm).Slop(100),
+				elastic.NewMatchPhraseQuery("authors.analyzed", exactTerm).Slop(100),
+			).MinimumNumberShouldMatch(1),
+		)
+	}
+	return elastic.NewFunctionScoreQuery().Query(query)
+	//.AddScoreFunc(elastic.NewGaussDecayFunction().FieldName("effective_date").Decay(0.9).Scale("300d"))
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go [hunk 28]
```diff
@@ -329,6 +329,83 @@ func AddCollectionsSearchRequests(mss *elastic.MultiSearchService, query Query,
...
+}
+
+func AddSourcesSearchRequests(mss *elastic.MultiSearchService, query Query, from int, size int, preference string) {
+	sources_indices := make([]string, len(query.LanguageOrder))
+	for i := range query.LanguageOrder {
+		sources_indices[i] = es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i])
+		log.Infof("adding index to sources_indices: %s", es.IndexName("prod", consts.ES_SOURCES_INDEX, query.LanguageOrder[i]))
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply search/engine.go]: log removed

File: search/engine.go (hunk 1 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 2 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 3 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 4 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 5 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 7 above)
Comment (YuriHechter) [reply search/engine.go]: this is checked inside joinResponses

File: search/engine.go (hunk 9 above)
Comment (YuriHechter) [reply search/engine.go]: log removed

File: search/engine.go (hunk 12 above)
Comment (YuriHechter) [reply search/engine.go]: Issue created AS-36

File: search/engine.go (hunk 13 above)
Comment (YuriHechter) [reply search/engine.go]: Issue created AS-36

File: search/engine.go (hunk 14 above)
Comment (YuriHechter) [reply search/engine.go]: Issue created AS-36

File: search/engine.go (hunk 8 above)
Comment (YuriHechter) [reply search/engine.go]: Issue created AS-36

File: search/engine.go (hunk 2 above)
Comment (YuriHechter) [reply search/engine.go]: not added to content.analyzed

File: es/common.go (hunk 15 above)
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/common.go]: changed to privateCase

File: es/indexer_test.go [hunk 29]
```diff
@@ -693,6 +699,191 @@ func deleteContentUnits(UIDs []string) error {
...
+		}
+		if source.Description != "" {
+			mdbSource.Description = null.NewString(source.Description, source.Description != "")
+		}
+		if err := mdbSource.Insert(common.DB); err != nil {
+			return "", err
+		}
+	}
+
+	if err := mdbSource.Update(common.DB); err != nil {
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: removed

File: es/indexer_test.go [hunk 30]
```diff
@@ -1134,3 +1405,54 @@ func (suite *IndexerSuite) TestCollectionsScopeByFile() {
...
+func (suite *IndexerSuite) TestSourcesIndex() {
+	fmt.Printf("\n\n\n--- TEST SOURCES INDEX ---\n\n\n")
+
+	r := require.New(suite.T())
+
+	fmt.Printf("\n\n\nAdding source.\n\n")
+	source1UID := suite.us(es.Source{Name: "test-name-1", Description: "test-description-1"}, consts.LANG_ENGLISH)
+	suite.us(es.Source{MDB_UID: source1UID, Name: "שם-בדיקה-1", Description: "תיאור-בדיקה-1"}, consts.LANG_HEBREW)
+	fmt.Printf("\n\n\nAdding content files for each language.\n\n")
+	suite.usfc(source1UID, consts.LANG_ENGLISH)
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: = done

File: es/indexer_test.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: added test to ensure that is not possible to index source without author

File: es/indexer_test.go [hunk 31]
```diff
@@ -1134,3 +1405,54 @@ func (suite *IndexerSuite) TestCollectionsScopeByFile() {
...
+	suite.validateSourceAuthors(indexNameEn, indexer, []string{"Test Name", "Test Full Name"})
+	suite.validateSourceAuthors(indexNameHe, indexer, []string{"שם לבדיקה", "שם מלא לבדיקה"})
+
+	fmt.Println("Remove 1 author and validate.")
+	suite.rsa(es.Source{MDB_UID: source1UID}, mdbmodels.Author{ID: 3})
+	r.Nil(indexer.SourceUpdate(source1UID))
+	suite.validateSourceAuthors(indexNameEn, indexer, []string{})
+
+	// Remove test indexes.
+	r.Nil(indexer.DeleteIndexes())
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: done

File: search/engine.go (hunk 4 above)
Comment (YuriHechter) [reply search/engine.go]: removed

File: es/sources_index.go [hunk 32]
```diff
@@ -0,0 +1,273 @@
...
+	if scope != emptyScope {
+		return index.Update(scope)
+	}
+	return nil
+}
+
+func (index *SourcesIndex) Update(scope Scope) error {
+	log.Infof("Sources Index - Update. Scope: %+v.", scope)
+	if scope.SourceUID != "" {
+		removed, err := index.removeFromIndexQuery(elastic.NewTermsQuery("mdb_uid", scope.SourceUID))
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: removeFromIndex added

File: es/sources_index.go [hunk 33]
```diff
@@ -0,0 +1,273 @@
...
+				//return err
+			}
+		}
+		offset += limit
+	}
+
+	return nil
+}
+
+func (index *SourcesIndex) removeFromIndexQuery(elasticScope elastic.Query) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/sources_index.go]: @bbfsdev in each index the function uses another type (from es\models.go) for json.Unmarshal. Possible to make this generic in go? Create generic interface for ES models?

File: es/indexer_test.go [hunk 34]
```diff
@@ -1134,3 +1405,54 @@ func (suite *IndexerSuite) TestCollectionsScopeByFile() {
...
+	// Index existing DB data.
+	r.Nil(indexer.ReindexAll())
+	r.Nil(indexer.RefreshAll())
+
+	fmt.Printf("\n\n\nValidate we have source with 2 languages.\n\n")
+	suite.validateSourceNames(indexNameEn, indexer, []string{"test-name-1"})
+	suite.validateSourceNames(indexNameHe, indexer, []string{"שם-בדיקה-1"})
+
+	fmt.Println("Validate source files.")
+	suite.validateSourceFile(indexNameEn, indexer, "TEST CONTENT")
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply es/indexer_test.go]: done

**Pull request closed without merging**